Core/include/Acts/EventData/detail/CurvilinearTrackParameters.hpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/detail/BoundTrackParameters.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+
+namespace Acts {
+namespace detail {
+
+/// Curvilinear track parameters for a single track.
+///
+/// @tparam particle_hypothesis_t Helper type to interpret the particle charge/momentum
+///
+/// This is intended as a user-facing data class that adds additional accessors
```

Comment:
So this parameter type went to `detail` because it is not user-facing? Then we should update the comment... But is it not user-facing anymore?

Reply:
This is exposed through `TrackParameters.hpp` so I would still say it is a user facing class but it should not be used via `detail::`

Reply:
I'd argue if the user is supposed to interact with it, it should not go into `detail`.

Reply:
okay - my only trouble with it is that the name then collides with the aliases in `TrackParameters.hpp`. so either I go back to `Single...` which I do not like too much or something like `...Impl` but then it sounds like a detail